unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/WorldChatWindowHUDController.cs

```diff
@@ -2,23 +2,23 @@
 using DCL;
 using DCL.Interface;
 using System.Collections;
+using System.Linq;
 using UnityEngine;
 
 public class WorldChatWindowHUDController : IHUD
 {
     private ChatHUDController chatHudController;
     public WorldChatWindowHUDView view;
 
-    private string userName;
-
     private IChatController chatController;
     private IMouseCatcher mouseCatcher;
 
     internal bool resetInputFieldOnSubmit = true;
+    private int invalidSubmitLastFrame = 0;
```

Comment:
are we using this somewhere?

Reply:
Yep, is to deal with a bug where `onSubmit` and the return key callback are called on the same frame, opening and closing the window again. We have an equivalent fix in the `UITextInput` UI component.